lewis_emulators/jsco4180/interfaces/stream_interface.py

```diff
@@ -73,94 +69,89 @@ def set_pump(self, mode):
             self.device.pump_mode = "Off"
             return self.out_terminator
 
+    @if_valid_input_delay
     @if_connected
+    @if_input_error
     def set_flowrate(self, flowrate):
         self.device.flowrate_sp = flowrate
         return self.out_terminator
 
+    @if_valid_input_delay
     @if_connected
     def get_flowrate(self):
         return self.device.flowrate
 
+    @if_valid_input_delay
     @if_connected
     def get_pressure(self):
         return self.device.pressure
 
+    @if_valid_input_delay
     @if_connected
     def set_pressure_max(self, pressure_max):
         self.device.pressure_max = pressure_max
         return self.out_terminator
 
+    @if_valid_input_delay
     @if_connected
     def get_pressure_max(self):
         return self.device.pressure_max
 
+    @if_valid_input_delay
     @if_connected
     def set_pressure_min(self, pressure_min):
         self.device.pressure_min = pressure_min
         return self.out_terminator
 
+    @if_valid_input_delay
     @if_connected
     def get_pressure_min(self):
         return self.device.pressure_min
 
-    @if_connected
-    def set_pump_off_timer(self, off_timer):
-        self.device.pump_off_timer = off_timer
-        return self.out_terminator
-
-    @if_connected
-    def set_pump_on_timer(self, on_timer):
-        self.device.pump_on_timer = on_timer
-        return self.out_terminator
-
-    @if_connected
-    def set_valve_position(self, valve_position):
-        self.device.valve_position = valve_position
-        return self.out_terminator
-
-    @if_connected
-    def get_valve_position(self):
-        return self.device.valve_position
-
-    @if_connected
-    def set_file_number(self, file_number):
-        self.device.file_number = file_number
-        return self.out_terminator
-
-    @if_connected
-    @if_input_error
-    def set_file_open(self, file_number):
-        self.device.file_open = True
-        self.device.file_number = file_number
-        return self.out_terminator
-
-    @if_connected
-    def set_file_close(self):
-        self.device.file_open = False
-
+    @if_valid_input_delay
     @if_connected
     def get_program_runtime(self):
         if self.device.pump_mode == 'Timed':
             self.device.program_runtime += 1
         return self.device.program_runtime
 
+    @if_valid_input_delay
     @if_connected
     def get_component_a(self):
-        return self.device.component_A
+        if self.device.single_channel_mode:
```

Comment:
Ternarys might be nice here `return 100 if self.device.single_channel_mode else self.device.component_A`